### PR TITLE
sbomnix: init at 1.4.5

### PR DESCRIPTION
**Pull request by henrirosten**
###### Description of changes

sbomnix is a utility that generates SBOMs from nix packages, for more details see: https://github.com/tiiuae/sbomnix

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by arianvp — APPROVED** [review]

**Comment by arianvp** [comment]
@GrahamcOfBorg build sbomnix

File: pkgs/top-level/all-packages.nix
Comment (06kellyjac) [inline pkgs/top-level/all-packages.nix:18762]:
```suggestion
  sbomnix = python3.pkgs.callPackage ../tools/security/sbomnix { };
```

This seems to be the preferred method for python applications

File: pkgs/tools/security/sbomnix/default.nix
Comment (06kellyjac) [inline pkgs/tools/security/sbomnix/default.nix:46]:
```suggestion
    maintainers = with maintainers; [ henrirosten jk ];
```

I'd be happy to maintain this with you if you'd like

File: pkgs/tools/security/sbomnix/default.nix
Comment (06kellyjac) [inline pkgs/tools/security/sbomnix/default.nix:37]:
```suggestion
    colorlog
    graphviz
    numpy
    packageurl-python
    pandas
    requests
    tabulate
```

After the change suggested in all-packages

File: pkgs/tools/security/sbomnix/default.nix
Comment (06kellyjac) [inline pkgs/tools/security/sbomnix/default.nix:12]:
```suggestion
{ lib
, fetchFromGitHub
, coreutils
, curl
, gnugrep
, gnused
, gzip
, graphviz
, nix
  # python libs
, reuse
, colorlog
, graphviz
, numpy
, packageurl-python
, pandas
, requests
, tabulate
}:
```

After the changes suggested in all-packages & the wrapper

**Review by 06kellyjac — COMMENTED** [review]
LGTM.

some minor comments

File: pkgs/top-level/all-packages.nix
Comment (henrirosten) [reply pkgs/top-level/all-packages.nix:18762]: Fixed in 0f1b2e7747bb9dea653ca76d107028bc25450b22

File: pkgs/tools/security/sbomnix/default.nix
Comment (henrirosten) [reply pkgs/tools/security/sbomnix/default.nix]: Thanks! Added you as maintainer in 0f1b2e7747bb9dea653ca76d107028bc25450b22

File: pkgs/tools/security/sbomnix/default.nix
Comment (henrirosten) [reply pkgs/tools/security/sbomnix/default.nix]: Fixed in 0f1b2e7747bb9dea653ca76d107028bc25450b22

File: pkgs/tools/security/sbomnix/default.nix
Comment (henrirosten) [reply pkgs/tools/security/sbomnix/default.nix]: Fixed in 0f1b2e7747bb9dea653ca76d107028bc25450b22, I hope I got your suggestion right.

**Comment by 06kellyjac** [comment]
Just noticed. the commit & title need the `v` prefix removed

Also it might be worth adding the other licenses used for the project to be fully accurate
https://github.com/tiiuae/sbomnix/tree/main/LICENSES

**Comment by henrirosten** [comment]
> Just noticed. the commit & title need the `v` prefix removed
> 
> Also it might be worth adding the other licenses used for the project to be fully accurate https://github.com/tiiuae/sbomnix/tree/main/LICENSES

Thanks, both fixed in 2e6ce810e49b093e27f8bb16846cc2436d352770. Also, fixed the PR title.

**Review by 06kellyjac — APPROVED** [review]

**Comment by henrirosten** [comment]
@06kellyjac : I see ofborg re-requested your review after your latest approval, probably because it now found you as a maintainer of sbomnix.

**Comment by 06kellyjac** [comment]
Ofborg can be a bit slow in requesting a review

**Review by 06kellyjac — APPROVED** [review]

**Comment by henrirosten** [comment]
Is there anything more I can do to help have this PR merged?

File: pkgs/tools/security/sbomnix/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/security/sbomnix/default.nix:9]:
```suggestion
, python
```
this fixes python version mixing

File: pkgs/tools/security/sbomnix/default.nix
Comment (SuperSandro2000) [inline pkgs/tools/security/sbomnix/default.nix:21]:
```suggestion
python.pkgs.buildPythonApplication rec {
```

**Review by teto — APPROVED** [review]
pretty cool software. I prefer to let previous reviewers merge.

File: pkgs/tools/security/sbomnix/default.nix
Comment (henrirosten) [reply pkgs/tools/security/sbomnix/default.nix]: Fixed in 8fae5ee3213024a3f498ebccbc1e94ac4a77ef8e

File: pkgs/tools/security/sbomnix/default.nix
Comment (henrirosten) [reply pkgs/tools/security/sbomnix/default.nix]: Fixed in 8fae5ee3213024a3f498ebccbc1e94ac4a77ef8e

**Pull request merged**
